### ISSUE TITLE
error in python script when i start up openshot

### ISSUE THREAD

**Issue by AndyIreland** [OpenShot/openshot-qt#2748]
![openshot error](https://user-images.githubusercontent.com/49172217/56426410-28377500-62b0-11e9-8eba-b963897f6170.jpg)

**Comment by ThomasBerends** [comment]
This looks like a duplicate of #1295 . Could you check if running OpenShot as Administrator solves the issue?

**Comment by Taurinus** [comment]
Exactly what I receive when launching this application.
Neither 32bit or 64bit, stable or latest beta,
always this Python error.
Other Python related programs are doing fine. Python 3.6 and Python3.7 "installed"....

**Comment by SuslikV** [comment]
Also, try to change the default installation path to the folder you have create by yourself "C:\OpenShot" for example.

**Comment by Taurinus** [comment]
So I changed the default installation path to somewhere in the root of my bootdrive, installed with admin rights.
Executed the application with admin rights, trying 32bit & 64bit version, always the same python error pops up.
Next thing I will try is to remove the installed Python 3.7 from windows path or even deinstall it.
And then I'm out of my guts ;-)...

**Comment by GeoffAdkins** [comment]
This error is still happening on Windows 7 32 bit, only Python 3.6 installed by OpenShot, no other Python as far as I can tell. Any fix yet ?

**Comment by SuslikV** [comment]
@GeoffAdkins this means that the openshot for some reason finds the file that itself is not part of the application and thus unable to load it properly. maybe the library was blocked by security software, or you have similar file in Windows system folder. For example, look at the system folder for any file named same as the any dll that is present in the _OpenShot Video Editor_ folder (look for Qt5..., avcodec..., avformat..., avutil..., swscale..., swresample... files first). Is there any of them in C:\Windows\System32\ folder?

**Comment by peanutbutterandcrackers** [comment]
Could you please take a look at this page and see if the work-arounds work for you: https://github.com/OpenShot/openshot-qt/wiki/Windows-Launch-Issues-(cx_Freeze-errors) ? Thank you!

**Comment by SuslikV** [comment]
At line 14 only libopenshot.dll is loads and fails, as far as I understand.

**Comment by Zipher04** [comment]
I have met the same error message on my Windows 10 x86.
I have tried the steps on https://www.openshot.org/issues/new/.
And the daily build(OpenShot-v2.4.4-dev2-1562263849-1b19ae7f-7939e156-x86) gave me a new error message which says "DLL load failed"
![2019-07-08_225037](https://user-images.githubusercontent.com/7078067/60819785-cada0280-a1d2-11e9-8fd7-29c67a81c75f.jpg)

I deleted the .openshot_qt folder and reopen the program. But the same error message shown again.

**Comment by SuslikV** [comment]
@Zipher04 Can you attach your recent `openshot-qt.log` file of OpenShot?

**Comment by Zipher04** [comment]
@SuslikV I checked the `openshot-qt.log` file. But it is blank. If there is anything I could help, please let me know.

![2019-07-10_233124](https://user-images.githubusercontent.com/7078067/60982504-d95b2200-a36a-11e9-917e-d94573322912.jpg)

**Comment by ferdnyc** [comment]
@Zipher04 Can you please try the latest daily build for x86, and see if you hit the same error? Our build process went a little awry for a few days, it's _possible_ the one you tried was still from that period. There are currently two more-recent 32-bit Windows daily builds listed at https://www.openshot.org/download/ .

**Comment by Zipher04** [comment]
@ferdnyc I tried the build `OpenShot-v2.4.4-dev2-1562710223-72b04ce8-7939e156-x86`. The still met the Import Error message box with a blank `openshot-qt.log`.
I tried to follow the message. Delete `.openshot-qt` folder and restart OpenShot. But still the same message was shown again.

**Comment by SuslikV** [comment]
@Zipher04 I just can't get this install to check if all files in place (probably connection issues). Meanwhile, what security software are you using? Can you look at its logs if some alerts was triggered for the OpenShot application during/after install?

**Comment by ferdnyc** [comment]
Hmm... strange. I downloaded that same installer, ran it in Wine on Linux, and I'm able to launch OpenShot with no issues. It reports an error opening the audio device (expected, since there's no real audio hardware under Wine), and the Preferences crash trying to detect Direct3D, but it does run.

![image](https://user-images.githubusercontent.com/538020/60986514-9fa7fe00-a30d-11e9-81ec-dff5f836e85d.png)

**Comment by ferdnyc** [comment]
OK, let's try this. Go into a Command Prompt, and run:
```command
cd c:\program files\openshot video editor\
launch.exe
```

See what's output on the terminal. (I'm assuming it'll be in "Program Files", since if you were on 64-bit Windows you'd be running the 64-bit build.)

**Comment by ferdnyc** [comment]
Actually... wait. _IS THERE_ a 32-bit version of Windows 10? I thought they were all 64-bit?

**Comment by ferdnyc** [comment]
Huh, shows what I know. https://www.newegg.com/microsoft-windows-10-pro-32-bit/p/N82E16832588521

**Comment by Zipher04** [comment]
I am sure I have a x86 windows 10. It is a ASUS transformer T100.
My anti-virus is Avast. I checked its log, it did not issue any alert.
Tried the command prompt, but still shown the same message box with no output.

![2019-07-11_003207](https://user-images.githubusercontent.com/7078067/60987086-7d48cb80-a373-11e9-859c-8ef7b3d9c274.jpg)
![2019-07-11_003034](https://user-images.githubusercontent.com/7078067/60987098-833eac80-a373-11e9-8d48-4b22a1f444ee.jpg)

**Comment by SuslikV** [comment]
@Zipher04 the files that can make similar issue are: `avformat-57.dll avcodec-57.dll avutil-55.dll swresample-2.dll swscale-4.dll libzmq.dll libopenshot-audio.dll libopenshot.dll` , if any is missing from the OpenShot installation folder - the yellow warning message will appear on the screen (the numbers before the `.dll` are installation specific and may vary).

**Comment by Zipher04** [comment]
@SuslikV I get all the .dll with exactly the same numbers you have mentioned.
![2019-07-11_004419](https://user-images.githubusercontent.com/7078067/60987911-2b08aa00-a375-11e9-93de-45817e0db7fe.jpg)

Really thank you for the kindly help. But it is late in night here. I will need to go for a sleep and will come back tomorrow.

**Comment by ferdnyc** [comment]
Yeah, it'll probably take us a few hours to puzzle this out anyway, I'm afraid. We may have some better ideas then. (We may not.)

I don't buy a missing file issue. Not through _three_ different installs of three different daily builds. There's just no way. It doesn't say there's necessarily a file missing. It says "the specified _**procedure**_ could not be found". What that implies to me is that it loaded a DLL, and found that it doesn't contain the expected set of symbols. Now, that's likely to be _either_:

* OpenShot packaged incorrectly, but with something that's only loaded on certain systems. I give that around a 25% likelihood.
* A system DLL getting in the way of one of the packaged DLLs, making a different set of symbols available. We've hit all sorts of examples of that kind of thing in the past. #2367 and the demon `zlib1.dll` comes instantly to mind.

So, the solution we eventually hit on there (installing Process Explorer, running OpenShot, then capturing a list of all the loaded DLLs at the point where that error is triggered) might provide some clues in this instance, too. My comment on #2367 detailing the data-collection procedure is [this one](https://github.com/OpenShot/openshot-qt/issues/2367#issuecomment-441472993).

**Comment by AndyIreland** [comment]
> I have met the same error message on my Windows 10 x86.
> I have tried the steps on https://www.openshot.org/issues/new/.
> And the daily build(OpenShot-v2.4.4-dev2-1562263849-1b19ae7f-7939e156-x86) gave me a new error message which says "DLL load failed"
> ![2019-07-08_225037](https://user-images.githubusercontent.com/7078067/60819785-cada0280-a1d2-11e9-8fd7-29c67a81c75f.jpg)
> 
> I deleted the .openshot_qt folder and reopen the program. But the same error message shown again.

same with me - win 7 x86

**Comment by ferdnyc** [comment]
@AndyIreland If you could also try the data-collection steps outlined in [my comment on #2367](https://github.com/OpenShot/openshot-qt/issues/2367#issuecomment-441472993), and post your Process Explorer DLL list, that might give us some clues as to what's going on here.

**Comment by ferdnyc** [comment]
The important bits there are...

- First, install Microsoft's [Process Explorer](https://docs.microsoft.com/en-us/sysinternals/downloads/process-explorer)

Then:
> - Run... OpenShot. (Don't worry about 64-bit.)
> - Get to the error dialog you've shown us, and leave it open
> - Go to (or launch) Process Explorer, and click on the launch.exe process
> - Click down inside the DLL list in the lower pane
> - Hit <kbd>Ctrl</kbd>+<kbd>a</kbd> which, for some reason, instead of doing a select-all, will open a file-save dialog offering to write a text file.
> - Save the text file, and then drag-and-drop it into a comment on this issue to upload it to GitHub.

**Comment by SuslikV** [comment]
@ferdnyc The x86 install just contains mess of x64 libs. Even `resvg.dll` is x64...

**Comment by AndyIreland** [comment]
here is error i get with x86 openshot: 
![openshot_error](https://user-images.githubusercontent.com/49172217/61007987-37393b00-a366-11e9-89c3-cf482ea102e1.jpg)

**Comment by ferdnyc** [comment]
My memory that we used to get better tracebacks from a lot of these problems has proved correct, that "Please delete this path that's completely unrelated to the problem you're having" popup was being overzealous about assuming every problem is a permission error. I've submitted #2862 to rein it in a lot, and hopefully restore more accurate/meaningful error reporting in more instances.

**Comment by ferdnyc** [comment]
> @ferdnyc The x86 install just contains mess of x64 libs. Even `resvg.dll` is x64...

Actually, it doesn't seem to be "a mess" of them, I just ran a quick scan of the installed OpenShot DLLs from the latest Windows 32-bit daily build, and `resvg.dll` is the _only_ one that detected as 64-bit. 

But, you're absolutely correct that the `resvg.dll` packaged in the Win32 daily builds is a 64-bit DLL, and that's definitely gonna cause some problems. Not necessarily the ones we've seen so far in this report, but problems for sure.

...Well, or maybe it means it'll just get ignored. I say that because it's the `freeze.py` script that adds the DLL to the build, and it's been making that same error _since January_. So, if it was completely crippling all of our Win32 builds, I'm sure we'd have heard a lot more about it long before now.

Still, it's definitely something that needs to be corrected. And that likely means @jonoomph first getting onto the private GitLab Windows builders and _creating_ a 32-bit ReSVG, so it can be installed into the x86 builds. Ping, @jonoomph ?

**Comment by ferdnyc** [comment]
@SuslikV Actually, looking at the symbols in the 32-bit `libopenshot.dll`, there are _no_ references to resvg whatsoever. So, it looks like 32-bit libopenshot might not be built with ReSVG support at all, and that 64-bit `resvg.dll` that's packaged in there is probably just a spurious file that's never actually used.

```console
$ cd "/path/to/64bit/OpenShot Video Editor"
$ x86_64-w64-mingw32-objdump -x libopenshot.dll |grep -i svg
	DLL Name: resvg.dll
	1124dc	    2  resvg_get_image_viewbox
	1124f8	    6  resvg_init_options
	112510	    9  resvg_parse_tree_from_data
	112530	   10  resvg_parse_tree_from_file
	112550	   12  resvg_qt_render_to_canvas
	11256c	   15  resvg_tree_destroy
	[  40] _ZN21QScopedPointerDeleterIN12ResvgPrivate4DataEE7cleanupEPS1_
[5034](sec  1)(fl 0x00)(ty  20)(scl   3) (nx 0) 0x000000000008c750 _ZN12ResvgPrivateL13errorToStringEi
[5056](sec  1)(fl 0x00)(ty   0)(scl   3) (nx 1) 0x00000000000c2db0 .text$_ZN21QScopedPointerDeleterIN12ResvgPrivate4DataEE7cleanupEPS1_
[5058](sec  1)(fl 0x00)(ty  20)(scl   2) (nx 0) 0x00000000000c2db0 _ZN21QScopedPointerDeleterIN12ResvgPrivate4DataEE7cleanupEPS1_
[5108](sec  5)(fl 0x00)(ty   0)(scl   3) (nx 1) 0x000000000000ad28 .xdata$_ZN21QScopedPointerDeleterIN12ResvgPrivate4DataEE7cleanupEPS1_
[5110](sec  4)(fl 0x00)(ty   0)(scl   3) (nx 1) 0x0000000000002760 .pdata$_ZN21QScopedPointerDeleterIN12ResvgPrivate4DataEE7cleanupEPS1_
[5144](sec 16)(fl 0x00)(ty   0)(scl   3) (nx 1) 0x0000000000013e98 .debug_frame$_ZN21QScopedPointerDeleterIN12ResvgPrivate4DataEE7cleanupEPS1_
[12220](sec  1)(fl 0x00)(ty  20)(scl   2) (nx 0) 0x00000000000c2ae8 resvg_init_options
[12341](sec  8)(fl 0x00)(ty   0)(scl   2) (nx 0) 0x0000000000002410 __imp_resvg_parse_tree_from_data
[12439](sec  8)(fl 0x00)(ty   0)(scl   2) (nx 0) 0x0000000000002400 __imp_resvg_get_image_viewbox
[12450](sec  1)(fl 0x00)(ty  20)(scl   2) (nx 0) 0x00000000000c2ae0 resvg_get_image_viewbox
[12660](sec  1)(fl 0x00)(ty  20)(scl   2) (nx 0) 0x00000000000c2af0 resvg_parse_tree_from_data
[12955](sec  8)(fl 0x00)(ty   0)(scl   2) (nx 0) 0x00000000000078c8 resvg_dll_iname
[12981](sec  8)(fl 0x00)(ty   0)(scl   2) (nx 0) 0x0000000000002420 __imp_resvg_qt_render_to_canvas
[12987](sec  1)(fl 0x00)(ty  20)(scl   2) (nx 0) 0x00000000000c2af8 resvg_parse_tree_from_file
[13064](sec  1)(fl 0x00)(ty  20)(scl   2) (nx 0) 0x00000000000c2b00 resvg_qt_render_to_canvas
[13149](sec  1)(fl 0x00)(ty  20)(scl   2) (nx 0) 0x00000000000c2b08 resvg_tree_destroy
[13255](sec  8)(fl 0x00)(ty   0)(scl   2) (nx 0) 0x0000000000002428 __imp_resvg_tree_destroy
[13558](sec  8)(fl 0x00)(ty   0)(scl   2) (nx 0) 0x0000000000002418 __imp_resvg_parse_tree_from_file
[13601](sec  8)(fl 0x00)(ty   0)(scl   2) (nx 0) 0x000000000000012c _head_resvg_dll
[13607](sec  8)(fl 0x00)(ty   0)(scl   2) (nx 0) 0x0000000000002408 __imp_resvg_init_options

$ cd "/path/to/32bit/OpenShot Video Editor"
$ i686-w64-mingw32-objdump -x libopenshot.dll |grep -i svg
[no results]
```

@AndyIreland Wow, that is _deep_ in the weeds of the dynamic loading system, nothing there even really gives a **hint** as to what it's actually trying to do.

**Comment by ferdnyc** [comment]
OK, #2862 has been merged, which means that starting from the next round of daily builds, the "delete your config folder" popup should **only** be shown in cases where there's a _permission_ problem that's preventing OpenShot from writing to its own configuration directory, meaning that it can't save any of the settings or working/temporary files.

(That's all it was intended to be doing, and the only case where the deletion advice is applicable — I guess it's a particular issue that's been cropping up more frequently on Win10 lately. But, because the permissions-error popup was catching lots of errors that _weren't_ permissions issues, it was blocking our ability to determine what errors are **actually** being encountered in all of those other cases.)

So, for those experiencing this issue, what I'll request is that, when the next round of daily builds show up (dated July 11 or later, which should be the ones with this fix included), you download and install _those_ versions of OpenShot, and then re-test this issue.

It's very unlikely that the new builds will **FIX** the problem you're experiencing — that's not the expectation here. What is expected to happen, though, is that they'll now provide some _actual_ error information regarding whatever problem you're experiencing, information that we can hopefully use to track down the issue. Rather than the permissions message popup that's just giving you wrong advice about deleting things that have nothing to do with your issue, and hiding the real problem.

Apologies for the confusion, hopefully we'll be able to get this sorted out soon.

**Comment by Zipher04** [comment]
@ferdnyc I checked the new daily build does not yet show up.
And I tried the Process Explorer method you provided. The text file is attached:
[launch.exe.txt](https://github.com/OpenShot/openshot-qt/files/3382416/launch.exe.txt)

**Comment by ferdnyc** [comment]
@Zipher04 Yeah, I'm not sure exactly how scheduling works on the build server. I'll keep an eye as well, and I'll take a look at the data you posted (which should be helpful anyway, it won't be affected by that pop-up issue), thanks.

**Comment by ferdnyc** [comment]
@Zipher04 Well, I was wrong. There are actually **no** DLLs listed in that file. So either something is interfering with DLL loading _entirely_, or it's a data-collection error.

Process Explorer is kind of a pain to work with, it's ancient software. Is it possible you missed step 4 in my list above (I've changed it from a bullet list to a numbered list, like it should've been in the beginning), "Click down inside the DLL list in the lower pane"? 

If you hit <kbd>Ctrl</kbd>+<kbd>a</kbd> without doing that first, then Process Explorer saved the wrong listing. I need the list of DLLs loaded, specifically.

**Comment by ferdnyc** [comment]
Here's an example of what the file's contents should look like, from @kazham's reply on that other bug. It's the bottom half that's the important part.

<details>
<summary>Contents of `launch.exe.txt` DLL list from Process Explorer</summary>

```
﻿Process	CPU	Private Bytes	Working Set	PID	Description	Company Name
armsvc.exe		1 224 K	4 296 K	1464	Adobe Acrobat Update Service	Adobe Systems Incorporated
ASROC.exe		6 408 K	13 164 K	2008		
atiesrxx.exe		1 716 K	4 660 K	636	AMD External Events Service Module	AMD
audiodg.exe	0.63	24 832 K	25 276 K	5764	Изоляция графов аудиоустройств Windows 	Microsoft Corporation
ClassicStartMenu.exe		4 744 K	10 800 K	128	Classic Start Menu	IvoSoft
cmd.exe		2 280 K	3 124 K	3108	Обработчик команд Windows	Microsoft Corporation
cmd.exe		2 316 K	3 768 K	6560	Обработчик команд Windows	Microsoft Corporation
cmon.exe	< 0.01	1 196 K	3 848 K	3276		
conhost.exe		2 008 K	6 300 K	4340	Окно консоли узла	Microsoft Corporation
conhost.exe	0.01	2 356 K	4 964 K	7600	Окно консоли узла	Microsoft Corporation
csrss.exe		2 188 K	4 680 K	584	Процесс исполнения клиент-сервер	Microsoft Corporation
csrss.exe	0.11	3 988 K	19 288 K	692	Процесс исполнения клиент-сервер	Microsoft Corporation
DbxSvc.exe		3 444 K	4 656 K	2452	Dropbox Service	Dropbox, Inc.
dwm.exe	0.23	28 548 K	44 244 K	1636	Диспетчер окон рабочего стола	Microsoft Corporation
EnergyController2.exe	0.34	63 280 K	1 760 K	1436	Energy Controller 2	Ибах О. И.
excavator.exe	0.58	448 752 K	14 080 K	4524		
explorer.exe	0.06	87 316 K	110 720 K	1692	Проводник	Microsoft Corporation
ffmpeg.exe	9.38	131 616 K	83 296 K	8464		
GoogleUpdate.exe		2 232 K	3 244 K	5028	Установщик Google	Google Inc.
iexplore.exe		12 968 K	31 024 K	2444	Internet Explorer	Microsoft Corporation
iexplore.exe		25 032 K	18 208 K	1284	Internet Explorer	Microsoft Corporation
Interrupts	16.52	0 K	0 K	n/a	Hardware Interrupts and DPCs	
launch.exe		42 404 K	23 808 K	4248		
Lingvo.exe	0.05	21 388 K	8 528 K	5572	Lingvo	ABBYY (BIT Software)
lsass.exe	0.05	5 588 K	13 468 K	804	Local Security Authority Process	Microsoft Corporation
lsm.exe		2 828 K	4 812 K	816	Служба диспетчера локальных сеансов	Microsoft Corporation
LvAgent.exe	0.01	1 784 K	6 444 K	5636	Lingvo Launcher	ABBYY (BIT Software)
MeGUI.exe	0.05	30 532 K	47 644 K	8956	MeGUI	http://sourceforge.net/projects/megui
NiceHash Miner 2.exe	0.02	32 856 K	14 288 K	3536	NiceHash Miner 2	H-BIT d.o.o.
NiceHash Miner 2.exe	0.29	274 588 K	33 480 K	2372	NiceHash Miner 2	H-BIT d.o.o.
NVDisplay.Container.exe		3 304 K	10 096 K	984	NVIDIA Container	NVIDIA Corporation
NVDisplay.Container.exe	< 0.01	24 224 K	39 412 K	1548	NVIDIA Container	NVIDIA Corporation
NvTelemetryContainer.exe		3 060 K	8 348 K	2668	NVIDIA Container	NVIDIA Corporation
OfficeClickToRun.exe	< 0.01	33 440 K	49 188 K	4400	Microsoft Office Click-to-Run (SxS)	Microsoft Corporation
Pen_Tablet.exe		2 120 K	6 116 K	2828	Tablet Service for consumer driver	Wacom Technology, Corp.
Pen_Tablet.exe	0.01	11 712 K	21 580 K	3016	Tablet Service for consumer driver	Wacom Technology, Corp.
Pen_TabletUser.exe		2 204 K	6 540 K	2948	Tablet user module for consumer driver	Wacom Technology, Corp.
Pen_TouchService.exe		1 976 K	5 400 K	1384	Touch Service	Wacom Technology, Corp.
Pen_TouchUser.exe	< 0.01	4 928 K	12 420 K	1604	Touch User Mode Driver	Wacom Technology, Corp.
procexp64.exe	1.34	32 172 K	49 800 K	8152	Sysinternals Process Explorer	Sysinternals - www.sysinternals.com
ps64ldr.exe	< 0.01	1 156 K	3 804 K	2312	Punto Switcher Loader for Win64	ООО Яндекс
punto.exe	0.09	11 592 K	19 804 K	1172	Punto Switcher	ООО Яндекс
SearchFilterHost.exe		2 624 K	7 124 K	7128	Microsoft Windows Search Filter Host	Microsoft Corporation
SearchIndexer.exe	0.02	30 984 K	19 036 K	4080	Индексатор службы Microsoft Windows Search	Microsoft Corporation
SearchProtocolHost.exe	< 0.01	3 512 K	9 144 K	7452	Microsoft Windows Search Protocol Host	Microsoft Corporation
services.exe		6 020 K	9 952 K	796	Приложение служб и контроллеров	Microsoft Corporation
smss.exe		556 K	1 348 K	340	Диспетчер сеанса  Windows	Microsoft Corporation
spoolsv.exe		7 124 K	12 724 K	1968	Диспетчер очереди печати	Microsoft Corporation
svchost.exe		5 160 K	10 892 K	920	Хост-процесс для служб Windows	Microsoft Corporation
svchost.exe	0.01	4 408 K	8 832 K	300	Хост-процесс для служб Windows	Microsoft Corporation
svchost.exe		21 912 K	18 444 K	1020	Хост-процесс для служб Windows	Microsoft Corporation
svchost.exe	< 0.01	11 200 K	20 452 K	1060	Хост-процесс для служб Windows	Microsoft Corporation
svchost.exe	0.04	31 628 K	47 420 K	1084	Хост-процесс для служб Windows	Microsoft Corporation
svchost.exe		2 672 K	6 240 K	1268	Хост-процесс для служб Windows	Microsoft Corporation
svchost.exe	< 0.01	7 960 K	16 280 K	1304	Хост-процесс для служб Windows	Microsoft Corporation
svchost.exe		6 540 K	9 940 K	2032	Хост-процесс для служб Windows	Microsoft Corporation
svchost.exe	< 0.01	18 692 K	16 672 K	1176	Хост-процесс для служб Windows	Microsoft Corporation
svchost.exe		2 380 K	7 328 K	2796	Хост-процесс для служб Windows	Microsoft Corporation
svchost.exe		1 972 K	6 060 K	2536	Хост-процесс для служб Windows	Microsoft Corporation
svchost.exe	< 0.01	3 712 K	7 420 K	4888	Хост-процесс для служб Windows	Microsoft Corporation
System	0.55	124 K	304 K	4		
System Idle Process	16.35	0 K	24 K	0		
taskeng.exe		2 556 K	7 156 K	2132	Обработчик планировщика заданий	Microsoft Corporation
taskhost.exe	0.05	16 764 K	16 372 K	1520	Хост-процесс для задач Windows	Microsoft Corporation
thunderbird.exe		155 480 K	188 420 K	4944	Thunderbird	Mozilla Corporation
TunnelBear.Maintenance.exe		21 972 K	38 144 K	3004	TunnelBear.Maintenance	
upwork.exe	1.26	96 976 K	34 632 K	4728	Upwork	Upwork, Inc.
upwork.exe		5 460 K	8 224 K	3336	Upwork	Upwork, Inc.
upwork.exe		68 352 K	11 220 K	1652	Upwork	Upwork, Inc.
upwork.exe	2.07	184 268 K	50 136 K	5464	Upwork	Upwork, Inc.
wininit.exe		1 716 K	4 972 K	700	Автозагрузка приложений Windows	Microsoft Corporation
winlogon.exe		3 472 K	8 124 K	748	Программа входа в систему Windows	Microsoft Corporation
wisptis.exe		4 464 K	9 332 K	1576	Компонент пера и сенсорного ввода (Майкрософт)	Microsoft Corporation
wisptis.exe	0.01	5 308 K	12 812 K	1584	Компонент пера и сенсорного ввода (Майкрософт)	Microsoft Corporation
WmiPrvSE.exe		2 808 K	7 224 K	4724	WMI Provider Host	Microsoft Corporation
x264.exe	49.88	683 456 K	519 816 K	3480	H.264 (MPEG-4 AVC) encoder	x264 project

Process: launch.exe Pid: 4248

Name	Description	Company Name	Path
_bz2-cpython-36m.dll			C:\Program Files\OpenShot Video Editor\lib\python3.6\_bz2-cpython-36m.dll
_ctypes-cpython-36m.dll			C:\Program Files\OpenShot Video Editor\lib\python3.6\_ctypes-cpython-36m.dll
_heapq-cpython-36m.dll			C:\Program Files\OpenShot Video Editor\lib\python3.6\_heapq-cpython-36m.dll
_lzma-cpython-36m.dll			C:\Program Files\OpenShot Video Editor\lib\python3.6\_lzma-cpython-36m.dll
_struct-cpython-36m.dll			C:\Program Files\OpenShot Video Editor\lib\python3.6\_struct-cpython-36m.dll
advapi32.dll	Расширенная библиотека API Windows 32	Microsoft Corporation	C:\Windows\System32\advapi32.dll
apisetschema.dll	ApiSet Schema DLL	Microsoft Corporation	C:\Windows\System32\apisetschema.dll
C_1252.NLS			C:\Windows\System32\C_1252.NLS
comctl32.dll	Библиотека элементов управления взаимодействия с пользователем	Microsoft Corporation	C:\Windows\winsxs\amd64_microsoft.windows.common-controls_6595b64144ccf1df_6.0.7601.23403_none_e36ad4593102f066\comctl32.dll
cryptbase.dll	Base cryptographic API DLL	Microsoft Corporation	C:\Windows\System32\cryptbase.dll
cryptsp.dll	Cryptographic Service Provider API	Microsoft Corporation	C:\Windows\System32\cryptsp.dll
dwmapi.dll	Интерфейс API диспетчера окон рабочего стола (Майкрософт)	Microsoft Corporation	C:\Windows\System32\dwmapi.dll
gdi32.dll	GDI Client DLL	Microsoft Corporation	C:\Windows\System32\gdi32.dll
imagehlp.dll	Windows NT Image Helper	Microsoft Corporation	C:\Windows\System32\imagehlp.dll
imageres.dll	Windows Image Resource	Microsoft Corporation	C:\Windows\System32\imageres.dll
imageres.dll.mui	Windows Image Resource	Microsoft Corporation	C:\Windows\System32\en-US\imageres.dll.mui
imm32.dll	Multi-User Windows IMM32 API Client DLL	Microsoft Corporation	C:\Windows\System32\imm32.dll
kernel32.dll	Библиотека клиента Windows NT BASE API	Microsoft Corporation	C:\Windows\System32\kernel32.dll
kernel32.dll.mui	Библиотека клиента Windows NT BASE API	Microsoft Corporation	C:\Windows\System32\ru-RU\kernel32.dll.mui
KernelBase.dll	Библиотека клиента Windows NT BASE API	Microsoft Corporation	C:\Windows\System32\KernelBase.dll
KernelBase.dll.mui	Библиотека клиента Windows NT BASE API	Microsoft Corporation	C:\Windows\System32\ru-RU\KernelBase.dll.mui
launch.exe			C:\Program Files\OpenShot Video Editor\launch.exe
libbz2-1.dll			C:\Program Files\OpenShot Video Editor\libbz2-1.dll
libffi-6.dll			C:\Program Files\OpenShot Video Editor\libffi-6.dll
libgcc_s_seh-1.dll			C:\Program Files\OpenShot Video Editor\libgcc_s_seh-1.dll
libicudt58.dll			C:\Program Files\OpenShot Video Editor\libicudt58.dll
libicuin58.dll			C:\Program Files\OpenShot Video Editor\libicuin58.dll
libicuuc58.dll			C:\Program Files\OpenShot Video Editor\libicuuc58.dll
liblzma-5.dll	liblzma data compression library	The Tukaani Project <http://tukaani.org/>	C:\Program Files\OpenShot Video Editor\liblzma-5.dll
libpcre2-16-0.dll			C:\Program Files\OpenShot Video Editor\libpcre2-16-0.dll
libpython3.6m.dll			C:\Program Files\OpenShot Video Editor\libpython3.6m.dll
libstdc++-6.dll			C:\Program Files\OpenShot Video Editor\libstdc++-6.dll
libwinpthread-1.dll	POSIX WinThreads for Windows	MingW-W64 Project. All rights reserved.	C:\Program Files\OpenShot Video Editor\libwinpthread-1.dll
locale.nls			C:\Windows\System32\locale.nls
lpk.dll	Language Pack	Microsoft Corporation	C:\Windows\System32\lpk.dll
mpr.dll	Библиотека маршрутизации для нескольких служб доступа	Microsoft Corporation	C:\Windows\System32\mpr.dll
msctf.dll	Серверная библиотека MSCTF	Microsoft Corporation	C:\Windows\System32\msctf.dll
msctf.dll.mui	Серверная библиотека MSCTF	Microsoft Corporation	C:\Windows\System32\ru-RU\msctf.dll.mui
msvcrt.dll	Windows NT CRT DLL	Microsoft Corporation	C:\Windows\System32\msvcrt.dll
netapi32.dll	Net Win32 API DLL	Microsoft Corporation	C:\Windows\System32\netapi32.dll
netutils.dll	Net Win32 API Helpers DLL	Microsoft Corporation	C:\Windows\System32\netutils.dll
nsi.dll	NSI User-mode interface DLL	Microsoft Corporation	C:\Windows\System32\nsi.dll
ntdll.dll	Системная библиотека NT	Microsoft Corporation	C:\Windows\System32\ntdll.dll
ole32.dll	Microsoft OLE для  Windows	Microsoft Corporation	C:\Windows\System32\ole32.dll
oleaut32.dll		Microsoft Corporation	C:\Windows\System32\oleaut32.dll
profapi.dll	User Profile Basic API	Microsoft Corporation	C:\Windows\System32\profapi.dll
pshook64.dll	Punto Switcher hook module	ООО Яндекс	C:\Program Files (x86)\Yandex\Punto Switcher\pshook64.dll
Qt5Core.dll	C++ Application Development Framework	The Qt Company Ltd.	C:\Program Files\OpenShot Video Editor\Qt5Core.dll
QtCore.pyd			C:\Program Files\OpenShot Video Editor\lib\python3.6\PyQt5\QtCore.pyd
rpcrt4.dll	Библиотека удаленного вызова процедур	Microsoft Corporation	C:\Windows\System32\rpcrt4.dll
rsaenh.dll	Microsoft Enhanced Cryptographic Provider	Microsoft Corporation	C:\Windows\System32\rsaenh.dll
sechost.dll	Host for SCM/SDDL/LSA Lookup APIs	Microsoft Corporation	C:\Windows\System32\sechost.dll
shell32.dll	Общая библиотека оболочки Windows	Microsoft Corporation	C:\Windows\System32\shell32.dll
shlwapi.dll	Библиотека небольших программ оболочки	Microsoft Corporation	C:\Windows\System32\shlwapi.dll
sip.pyd			C:\Program Files\OpenShot Video Editor\lib\python3.6\sip.pyd
SortDefault.nls			C:\Windows\Globalization\Sorting\SortDefault.nls
srvcli.dll	Server Service Client DLL	Microsoft Corporation	C:\Windows\System32\srvcli.dll
StaticCache.dat			C:\Windows\Fonts\StaticCache.dat
user32.dll	Многопользовательская библиотека клиента USER API Windows	Microsoft Corporation	C:\Windows\System32\user32.dll
userenv.dll	Userenv	Microsoft Corporation	C:\Windows\System32\userenv.dll
usp10.dll	Uniscribe Unicode script processor	Microsoft Corporation	C:\Windows\System32\usp10.dll
uxtheme.dll	Библиотека тем UxTheme (Microsoft)	Microsoft Corporation	C:\Windows\System32\uxtheme.dll
version.dll	Version Checking and File Installation Libraries	Microsoft Corporation	C:\Windows\System32\version.dll
winmm.dll	MCI API DLL	Microsoft Corporation	C:\Windows\System32\winmm.dll
wkscli.dll	Workstation Service Client DLL	Microsoft Corporation	C:\Windows\System32\wkscli.dll
ws2_32.dll	32-разрядная библиотека Windows Socket 2.0	Microsoft Corporation	C:\Windows\System32\ws2_32.dll
zlib1.dll	zlib data compression library		C:\Windows\System32\zlib1.dll
zlib-cpython-36m.dll			C:\Program Files\OpenShot Video Editor\lib\python3.6\zlib-cpython-36m.dll
```
</details>

**Comment by ferdnyc** [comment]
@AndyIreland (I haven't forgotten that this was originally your bug report): 

Like I said a few comments back, that traceback in your screenshot is _impressively_ devoid of actual information on what it's trying (and failing) to do. The only suggestion I have is to try the same DLL check. I'm not as confident it will provide the info we need, but it's the only thing I can think of. So, same steps:

1. Install Microsoft's [Process Explorer](https://docs.microsoft.com/en-us/sysinternals/downloads/process-explorer) tool.
1. Run OpenShot.
1. Get to the error dialog you've shown us, and leave it open
1. Go to (or launch) Process Explorer, and click on the `launch.exe` process
1. Click down inside the DLL list in the lower pane
1. Hit <kbd>Ctrl</kbd>+<kbd>a</kbd> which, for some reason, instead of doing a select-all, will open a file-save dialog offering to write a text file.
1. Save the text file, and then drag-and-drop it into a comment on this issue to upload it to GitHub.

**Comment by AndyIreland** [comment]
thanks. I have clicked on launcher .exe in the process explorer tool and highlighted it , but I dont get what to do next - step number 5  .. where is this click down inside the DLL list in the lower pane ? - I dont see it

**Comment by ferdnyc** [comment]
Hmmm. Possibly Process Explorer doesn't _show_ the DLL list by default. (It's been so long since I've run it, I can hardly remember. If that's the case, sorry, @Zipher04 !) Is there a Window or View menu option to open a lower pane in the window, and possibly to select a list of loaded DLLs as its contents? I'll see if I can find the help file or something online, figure out what I'm half-remembering.

**Comment by ferdnyc** [comment]
Aha. According to the help file,

> The **Process Explorer** window shows by default two panes: the upper pane is always a process list and the bottom either shows the list of DLLs loaded into the process selected in the upper pane, or the list of operating system resource handles (files, Registry keys, synchronization objects) the process has open; the view mode determines which information is shown in the bottom pane. To switch the view, use the **View|Lower Pane View** menu item, the corresponding toolbar button (which toggles), or the Ctrl+D (DLL view) and Ctrl-H (handle view) accelerator keys.

So you'd want to hit <kbd>Ctrl</kbd>+</kbd>d</kbd>, or use the View menu, to set the lower pane to show a list of DLLs loaded by the process selected in the upper pane. Then, you'd click anywhere in that (lower-pane) list and hit <kbd>Ctrl</kbd>+</kbd>a</kbd> to save the list.

**Comment by ferdnyc** [comment]
And just to add to that, in case it's useful, the next paragraph from the help file:

> If you are only interested in seeing the processes running on your system You can hide the lower pane by selecting **View|Hide Lower Pane**, the corresponding toolbar button, the Ctrl+L accelerator, or by dragging the pane divider to the bottom of the Process Explorer window. You can bring back the lower pane by selecting **View|Show Lower Pane**, typing Ctrl+L or selecting the toolbar button again.

**Comment by Taurinus** [comment]
I dont know exactly what has happened...
But installing the latest shotcut 64bit beta and/or updating to python 3.7.4 64bit solved the launch errors.
So for me, case closed ;-)

**Comment by Zipher04** [comment]
@AndyIreland The lower panel of .dll can be found here.
![2019-07-13_224642](https://user-images.githubusercontent.com/7078067/61173037-34d21d80-a5c0-11e9-922e-7bc7c3ff8be6.jpg)

@ferdnyc Sorry, I did not notice the .dll does not in Process Explorer by default. I have update the list of .dll as the following: [launch.exe.txt](https://github.com/OpenShot/openshot-qt/files/3388902/launch.exe.txt)

**Comment by Zipher04** [comment]
@ferdnyc I just tried the new daily build `OpenShot-v2.4.4-dev2-1562913165-ed4b32d4-7939e156-x86.exe`. But still met the same message:
![2019-07-14_000654](https://user-images.githubusercontent.com/7078067/61173864-90ee6f00-a5cb-11e9-9642-d0ea9c62afd2.jpg)

The list from Process Explorer for this version is also attached: [launch.exe.txt](https://github.com/OpenShot/openshot-qt/files/3388969/launch.exe.txt).

**Comment by SuslikV** [comment]
@Zipher04 , Do you have something written into `openshot-qt.log` file or it is still blank?

**Comment by ferdnyc** [comment]
Arrgh, there's a _second_ one of those? Well, that explains it. I only fixed the permission-error one. **That** box has to go, it's hiding all the useful data. I'll take care of it later, I promised myself I'd see at least SOME sunlight today.

> @ferdnyc Sorry, I did not notice the .dll does not in Process Explorer by default. I have update the list of .dll as the following

@Zipher04 Not to worry, thanks. Still some issues on our end, I'm afraid, so stay tuned. The info you sent so far doesn't show anything out of the ordinary (no DLLs being loaded from odd places, like I feared) so we'll have to get rid of that popup message first, before we can access any additional information about the problem.

**Comment by ferdnyc** [comment]
OK, I just merged #2867 which _completely_ addresses this issue, so OpenShot should now produce **much** better information in the event of any problems. You can see the pull request for details if you like, but basically the popup for import errors will now show the exception traceback, instead of that poor advice about deleting your configuration (which has no hope of solving an import error).

Next round of daily builds should be useful for finally solving this, whenever they show up. The latest ones there right now are three (...? Weird number!) dated July 11, so any newer than that will contain this fix. Stay tuned...

**Comment by SuslikV** [comment]
I think, that user should manually delete the OpensShot's installation folder after application uninstall and PC restart. And only then you should install recent build of the OpenShot again.

**Comment by ferdnyc** [comment]
I'd rather see error details before giving any sort of advice. If they do that now, even if it fixes the problem, we lose the chance to see a traceback that might let us pinpoint the cause of this problem so we can prevent it happening in the future.

**Comment by ferdnyc** [comment]
(But, yes, if anyone has an urgent need that can't wait for our troubleshooting attempts, completely wiping out any trace of OpenShot, then installing from scratch — I'd suggest 2.4.4 release, rather than the Daily Builds — _might_ solve the problem. And if it doesn't, then you've lost nothing really, and we can still hopefully get the information we need as of the next Daily Build release.)

**Comment by SuslikV** [comment]
Let me guess @ferdnyc - the libopenshot fails to load... OK let's wait for the results if any will be.

**Comment by Zipher04** [comment]
@ferdnyc I have tried the new daily build `OpenShot-v2.4.4-dev2-1563113590-ed4b32d4-7939e156-x86.exe`. The new error message is quite like the original one, but start with an additional line `Module: _openshot`.
![2019-07-14_231119](https://user-images.githubusercontent.com/7078067/61185502-e474c180-a68c-11e9-9821-83bebb7d39a6.jpg)

**Comment by Zipher04** [comment]
After I clicked OK. There was a second error message: 
![2019-07-14_231340](https://user-images.githubusercontent.com/7078067/61185526-17b75080-a68d-11e9-982c-6a37dec3f046.jpg)

**Comment by SuslikV** [comment]
@Zipher04 The other solution is to extract OpenShot files to any new empty folder you made and run the `launch.exe` file manually. The _innoextract_ command line util for Windows may help to extract all required files from the OpenShot installation. It may work. At least it works for Win7 x86 without issues. You may try it.

**Comment by ferdnyc** [comment]
I've asked @jonoomph to forward build logs from the most recent Daily Build for Windows 32-bit, I'll look over those once I have them and see if I can spot anything that looks off.

Unfortunately I don't have access to a Windows 10 32-bit system (or any Windows 10 system), which makes this difficult to debug. Everything works fine in my 64-bit Windows 7 VM, though I doubt that helps any.

It's _vaguely_ possible we're hitting a version of [this nasty bug](https://github.com/python-pillow/Pillow/issues/2479), one that [bit everyone](https://bugs.python.org/issue29943) two years ago during the Python 3.6.0 to 3.6.1 transition. OpenShot _does_ get built and bundled with Python 3.6 on Windows. (I'm not exactly sure _which_ Python 3.6 release, though, which is one of many things I'm hoping the build logs will tell me.)

I'm not **at all** certain that's what's going on here, but the symptoms are the same — right down to the tracebacks that conspicuously lack any mention of exactly _what_ "specified procedure" they're not finding. Still, I assume a change in the Python library isn't the **only** possible cause of that message, so I don't want to assume anything at this point.

**Comment by Zipher04** [comment]
@SuslikV I tried to install `OpenShot-v2.4.4-x86.exe` into a folder on my desktop, but still met the same error message.
![2019-07-15_225427](https://user-images.githubusercontent.com/7078067/61225622-8232c400-a753-11e9-9249-fdd993655b00.jpg)

**Comment by SuslikV** [comment]
@Zipher04 Did you unpacked it as I suggested (the _innoextract_ util) or you just run the installer? Also, please, make simple folder like `C:\OpenShot` or even different drive, not the desktop folder. If you have USB-stick (SD card etc.) you can try to unpack OpenShot installation to that drive as well and run OpenShot from it.

**Comment by Zipher04** [comment]
@SuslikV Thank you for explained it to me again. I did not notice that `innoextract` is a utility I need to download.
I tried to extract `OpenShot-v2.4.4-x86.exe` by innoextract in command line today. But the "cx_Freeze: Python error in main script" error still shown.
Anyway, thanks for the information.

**Comment by SuslikV** [comment]
@Zipher04 the _libopenshot_ and related files are blocked in your PC. Look for OS logs. Because the mentioned installation works without mentioned issues. Either security reasons or lack of permissions. Of course, it is only my thoughts on your problem. Anyway, you spend a lot of time on solving this so, just try another software solutions for video edit...

**Comment by Zipher04** [comment]
@ferdnyc I tried the new daily build `OpenShot-v2.4.4-dev2-1564165823-769efd87-574877b9-x86`. And this time, the error message is gone, instead a Windows screen asking for network permission was shown. After that, OpenShot is finally open on my PC.
Thanks a lot for the inspections and the hard works!

**Comment by ferdnyc** [comment]
@Zipher04 That's great news! All credit goes to @jonoomph who spent some time digging deep into the build process, where he discovered some DLL ordering issues that were subtly (and stealthily) screwing us over.

If you're using the _very_ newest Daily Builds, there may be a _different_ issue, though. If your OpenShot window looks like this:

![image](https://user-images.githubusercontent.com/538020/61998952-803b0000-b085-11e9-8cac-8599fef821dc.png)

instead of looking more like this:

![image](https://user-images.githubusercontent.com/538020/61998954-88933b00-b085-11e9-9ced-0be11bc7ac34.png)

Then the icon-rendering issue reported in #2894 is affecting the build you've installed, as well.

However, if that's the case, then I _believe_ I can offer One Weird Trick™ to fix your installed build,  without having to downgrade or jump through too many hoops. I just have to track down a DLL and upload it here, shouldn't take more than a couple of minutes...

**Comment by ferdnyc** [comment]
Actually, it looks like I don't even need to do that, the necessary files are probably already installed, just in the wrong place.

Experimentally, at least, the fix for that icon issue (if you're seeing the icon issue) may be as easy as going into `C:\Program Files\OpenShot Video Editor\PyQt5\`, and **copying** (not moving) every `Qt*.dll` file up one level, to `C:\Program Files\OpenShot Video Editor\`

**Comment by ferdnyc** [comment]
It's been reported by @nanohtpc in #2894 that my suggested workaround (copying the Qt DLLs) is effective in fixing the issues with our most recent Daily Builds.

**Comment by ferdnyc** [comment]
And now the issue I noted in my previous two or three comments should be fixed, at least as of the next Daily Builds the servers generate.

**Comment by Zipher04** [comment]
@ferdnyc  The icon was also missing on my case. And it is fixed by the method you provided. Thank you for the information.

P.S. I just finished my first video project today. Really thank you for all the helps.

**Comment by ferdnyc** [comment]
@Zipher04 Glad to hear it, happy we could get things sorted out for you! 

Yeah, there was a hiccup in the packaging process when the project moved to new build servers, but everything should be back to normal (actually, better-than-normal, compared to previously) since @jonoomph fixed things up in #2891 / #2897.

Closing as fixed via the reworked packaging code.
